src/generic/UnivPoly.jl

```diff
@@ -10,11 +10,11 @@
 #
 ###############################################################################
 
-base_ring_type(::Type{<:UniversalPolyRing{T}}) where T = parent_type(T)
-base_ring(S::UniversalPolyRing) = base_ring(mpoly_ring(S))::base_ring_type(S)
+base_ring_type(::Type{<:UniversalPolyRing{T}}) where T = mpoly_ring_type(T)
+base_ring(S::UniversalPolyRing) = S.mpoly_ring::base_ring_type(S)
```

Comment:
As we just discussed maybe rename the field `mpoly_ring` to `base_ring` in preparation for your future `UniversalRing` changes.